src/react/projects/spark-react/src/SprkInput/components/SprkTextareaCheck.js

```diff
@@ -11,6 +11,16 @@ class SprkTextareaCheck extends Component {
     };
   }
 
+  componentDidMount() {
+    if (this.props.type === 'hugeTextInput') {
+      if (this.props.value !== '') {
+        this.setState({
+          hasValue: true,
+        });
+      }
+    }
+  }
+
```

Comment:
i dont like that SprkTextareaCheck is the owner of this state. I think it'd be better as part of SprkTextInput.

Reply:
Thanks Rob. Why does it make more sense to have this in SprkTextInput?